ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -598,70 +592,19 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 		} else if diff == 0 {
 			r.Log.Info("reconcilePods", "all workers already exist for group", worker.GroupName)
 			continue
-		} else if -diff == int32(len(worker.ScaleStrategy.WorkersToDelete)) {
-			r.Log.Info("reconcilePods", "removing all the pods in the scaleStrategy of", worker.GroupName)
-			for _, podsToDelete := range worker.ScaleStrategy.WorkersToDelete {
-				pod := corev1.Pod{}
-				pod.Name = podsToDelete
-				pod.Namespace = utils.GetNamespace(instance.ObjectMeta)
-				r.Log.Info("Deleting pod", "namespace", pod.Namespace, "name", pod.Name)
-				if err := r.Delete(ctx, &pod); err != nil {
-					if !errors.IsNotFound(err) {
-						return err
-					}
-					r.Log.Info("reconcilePods", "workers specified to delete was already deleted ", pod.Name)
-				}
-				r.Recorder.Eventf(instance, corev1.EventTypeNormal, "Deleted", "Deleted pod %s", pod.Name)
-			}
-			continue
 		} else {
-			// diff < 0 and not the same absolute value as int32(len(worker.ScaleStrategy.WorkersToDelete)
-			// we need to scale down
-			workersToRemove := int32(len(runningPods.Items)) - workerReplicas
-			randomlyRemovedWorkers := workersToRemove - int32(len(worker.ScaleStrategy.WorkersToDelete))
-			// we only need to scale down the workers in the ScaleStrategy
-			r.Log.Info("reconcilePods", "removing all the pods in the scaleStrategy of", worker.GroupName)
-			for _, podsToDelete := range worker.ScaleStrategy.WorkersToDelete {
-				pod := corev1.Pod{}
-				pod.Name = podsToDelete
-				pod.Namespace = utils.GetNamespace(instance.ObjectMeta)
-				r.Log.Info("Deleting pod", "namespace", pod.Namespace, "name", pod.Name)
-				if err := r.Delete(ctx, &pod); err != nil {
+			// diff < 0 means that we need to delete some Pods to meet the desired number of replicas.
+			randomlyRemovedWorkers := -diff
+			for i := 0; i < int(randomlyRemovedWorkers); i++ {
+				randomPodToDelete := runningPods.Items[i]
+				r.Log.Info("Randomly deleting Pod ", "index ", i, "/", randomlyRemovedWorkers, "with name", randomPodToDelete.Name)
```

Comment:
nit: `i` is based on 0 index and `randomlyRemovedWorkers` is actual count. 

Lets say `randomlyRemovedWorkers` = 10 and if you are removing last worker, this log will print `9/10`.